### PR TITLE
week 3 homework done

### PR DESCRIPTION
**Pull request by Serafettin99**

### CODE REVIEW

File: Week3/homework/cc-validator.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+const checkCharAmount = function(creditCardNum) {
+  if (creditCardNum.length === 16) {
+    return entries.valid;
+  }
+  return entries.invalid;
+};
+
+//all of them must be numbers (Indeed, the checkEven and checkSum funcs satisfy the order, nevertheless i have coded for it)
+const checkChars = function(creditCardNum) {
+  if (isNaN(creditCardNum)) {
```
Comment (nudge) [inline Week3/homework/cc-validator.js:42]: Nice use of `isNaN()`!

File: Week3/homework/cc-validator.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+};
+
+// Function expression
+let validateTheNumber = function(
+  cardNumber,
+  charAmountCallback,
+  charsCallback,
+  evenCallback,
+  sumCallback,
+  sameDigitsCallback,
```
Comment (nudge) [inline Week3/homework/cc-validator.js:91]: Pretty cool use of callbacks!
To make this really super amazing you could pass in an array of functions, then the function body can just loop over the array and call each function to make the code really compact. Nonetheless, this code works well, nice work!

File: Week3/homework/cc-validator.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+console.log(
+  validateTheNumber(
+    '6666666666666661',
+    checkCharAmount,
+    checkChars,
+    checkEven,
+    checkSum,
+    checkSameDigits,
+  ),
+);
```
Comment (nudge) [inline Week3/homework/cc-validator.js:125]: 👍

File: Week3/homework/exercise-1.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    'hardworking',
+    'marvelous',
+    'sweet',
+  ];
+  let myCompliment = `You are ${
+    compliments[Math.floor(Math.random() * compliments.length)]
+  } ${myName}`;
+  return myCompliment;
+}
+console.log(giveCompliment('Serafettin'));
```
Comment (nudge) [inline Week3/homework/exercise-1.js:31]: 👍 Nice use of string templating!

File: Week3/homework/exercise-2.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+// Return a string: "Your doggie is [CALCULATED_VALUE] years old in dog years!"
+// Call the function three times with different sets of values.
+function calculateDogAge(puppysAge) {
+  let calculatedAge = puppysAge * 7;
+  return `Your doggie is ${calculatedAge} years old in dog years!`;
+}
+
+console.log(calculateDogAge(2));
+console.log(calculateDogAge(3));
+console.log(calculateDogAge(1));
```
Comment (nudge) [inline Week3/homework/exercise-2.js:18]: 👍

File: Week3/homework/exercise-3.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+// Randomly select values from the arrays.
+// Return a string: "You will be a [JOB_TITLE] in [LOCATION], and married to [PARTNER_NAME] with [NUMBER_KIDS] kids."
+// Create 4 arrays, numChildren, partnerNames, locations and jobs. Give each array 5 random values that make sense
+// Call the function 1 time, by passing the arrays as the argument.
+
+//i need a function that selects random number so i've declared it.
+function randomElement(array) {
+  let randomIndex = Math.floor(Math.random() * array.length);
+  return array[randomIndex];
+}
```
Comment (nudge) [inline Week3/homework/exercise-3.js:17]: Nice work splitting this into a separate function! You can really see how using functions allows you to write re-usable code.

File: Week3/homework/exercise-3.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  let randomJob = randomElement(jT);
+  return `You will be a ${randomJob} in ${randomLoc}, and married with ${randomPartner} with ${randomChildNmr} kids.`;
+}
+
+let numChildren = [1, 2, 3, 4, 5];
+let partnerNames = ['Adriana', 'Johanna', 'Paulina', 'Messi', 'Ronaldo'];
+let locations = ['Amsterdam', 'Hilversum', 'Amesfoort', 'Nijmegen', 'Zutphen'];
+let jobs = ['doctor', 'dentist', 'worker', 'web - developer', 'conductor'];
+
+console.log(tellFortune(numChildren, partnerNames, locations, jobs));
```
Comment (nudge) [inline Week3/homework/exercise-3.js:34]: 👍

File: Week3/homework/exercise-4.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  if (listOfGroceryItems.length > 3) {
+    listOfGroceryItems.shift();
+  }
+  return listOfGroceryItems;
+}
+let listOfGroceryItems = ['bananas', 'milk'];
+addToShoppingCart('apple');
+addToShoppingCart('grapes');
+addToShoppingCart('watermeloen');
+console.log(listOfGroceryItems);
```
Comment (nudge) [inline Week3/homework/exercise-4.js:23]: :+1:

File: Week3/homework/exercise-5.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+const cartForParty = {
+  colas: 10,
+  chips: 20,
+  biscuits: 25,
+  cakes: 30,
+  dragees: 12,
+};
+
+console.log(calculateTotalPrice(cartForParty));
```
Comment (nudge) [inline Week3/homework/exercise-5.js:29]: :+1:

File: Week3/homework/cc-validator.js (hunk 1 above)
Comment (Serafettin99) [reply Week3/homework/cc-validator.js:42]: Thanks

File: Week3/homework/cc-validator.js (hunk 2 above)
Comment (Serafettin99) [reply Week3/homework/cc-validator.js:91]: i'll try. thanks

File: Week3/homework/exercise-1.js (hunk 4 above)
Comment (Serafettin99) [reply Week3/homework/exercise-1.js:31]: thank you

File: Week3/homework/exercise-3.js (hunk 6 above)
Comment (Serafettin99) [reply Week3/homework/exercise-3.js:17]: Yes i 've learned it. Thank you

**Comment by Serafettin99** [comment]
@nudge, I am grateful for your review and feedbacks. I've learned a lot thanks to you and of course @yash-kapila . Thank you both.

**Pull request closed without merging**